src/test/java/itest/RulesPostMultipartIT.java

```diff
@@ -0,0 +1,140 @@
+                .delete(String.format("/api/v2/rules/%s", TEST_RULE_NAME))
+                .send(
+                        ar -> {
+                            if (ar.succeeded()) {
+                                deleteResponse.complete(ar.result().bodyAsJsonObject());
+                            } else {
+                                deleteResponse.completeExceptionally(ar.cause());
+                            }
+                        });
+
+        JsonObject expectedDeleteResponse =
+                new JsonObject(
+                        Map.of(
+                                "meta",
+                                Map.of("type", HttpMimeType.JSON.mime(), "status", "OK"),
+                                "data",
+                                NULL_RESULT));
+
+        try {
+            JsonObject deleteResult = deleteResponse.get(5, TimeUnit.SECONDS);
+            MatcherAssert.assertThat(deleteResult, Matchers.equalTo(expectedDeleteResponse));
+        } catch (Exception e) {
+            System.out.println("Deletion failed. Reason: " + e.getMessage());
+            throw e;
+        }
+    }
+
+    @ParameterizedTest
+    @ValueSource(
+            strings = {
+                "multipart/form-data",
+                "multipart/form-data; boundary=------somecharacters",
+                "multipart/form-data; unkown characters",
+                "multipart/form-data; directive1; directive2",
+                "multipart/form-data; directive"
+            })
+    void shouldAcceptMultipartWithBoundary(String contentType) throws Exception {
+        MultiMap form = MultiMap.caseInsensitiveMultiMap();
+        form.set("name", TEST_RULE_NAME);
+        form.set("matchExpression", "false");
+        form.set("eventSpecifier", "template=Continuous");
+
+        CompletableFuture<JsonObject> response = new CompletableFuture<>();
+        webClient
+                .post("/api/v2/rules")
+                .putHeader(HttpHeaders.CONTENT_TYPE.toString(), contentType)
+                .sendForm(
+                        form,
+                        ar -> {
+                            if (assertRequestStatus(ar, response)) {
+                                response.complete(ar.result().bodyAsJsonObject());
+                            } else {
+                                response.completeExceptionally(
+                                        new RuntimeException("Request failed"));
+                            }
+                        });
+        try {
+            JsonObject result = response.get(5, TimeUnit.SECONDS);
+            System.out.println("Received response: " + result.toString());
+
+            MatcherAssert.assertThat(result.getJsonObject("meta"), Matchers.notNullValue());
+            MatcherAssert.assertThat(
+                    result.getJsonObject("meta").getString("status"), Matchers.equalTo("Created"));
+            MatcherAssert.assertThat(result.getJsonObject("data"), Matchers.notNullValue());
```

Comment:
Might it be worth checking the inner contents of this `data` field to ensure that the response matches the form submitted by the request?

Reply:
for this what Understood is i should just check the existence of the meta field and remove the data?

Reply:
No, I mean continue checking the data. But check more than just `notNullValue`. Check the contents of `data`, too.

Reply:
did i get this  right?
 MatcherAssert.assertThat(
                result.getJsonObject("data").getString("name"), Matchers.equalTo(TEST_RULE_NAME));
            MatcherAssert.assertThat(
                result.getJsonObject("data").getString("matchExpression"), Matchers.equalTo("false"));
            MatcherAssert.assertThat(
                result.getJsonObject("data").getString("eventSpecifier"), Matchers.equalTo("template=Continuous"));

Reply:
I mean this is what you mean?

Reply:
Sure, that looks good.

Reply:
i can add one that checks the contentType as well                                                                                                                             result.getJsonObject("data").getString("contentType"), Matchers.equalTo(contentType));